### ISSUE TITLE
No sensors in HA

### ISSUE THREAD

**Issue by mist0706** [GMTrevis/Homeassistant-NodeRed-Systemair-VTR300#5]
Unsure of what i'm doing wrong but none of the sensors seem to be showing up in HA even though i've put in the correct ip and most likely followed all of the instructions

![Screenshot 2022-06-02 at 09 50 19](https://user-images.githubusercontent.com/10143260/171581280-45da53ff-a2a9-419d-9a60-a365b713c98c.png)

**Comment by GMTrevis** [comment]
> Unsure of what i'm doing wrong but none of the sensors seem to be showing up in HA even though i've put in the correct ip and most likely followed all of the instructions `![Screenshot 2022-06-02 at 09 50 19](https://user-images.githubusercontent.com/10143260/171581280-45da53ff-a2a9-419d-9a60-a365b713c98c.png)`

Can you read data from the modbusregisters in Node red?
If not:
 - Is the IAM gateway configured as "Modbus Gateway", ref. picture 1 below.
 - Is the modbus unit slave adr. and "unitid" in the "Modbus read" function nodes in Node red equal? Ref. picture 2 below.

I assume you already have the node red companion integration installed? Ref. picture 3.
If you read modbus data in Node red, i cannot think of much that could cause modbus data not to be displayed in HA. 

Picture 1:
![IAM gateway](https://user-images.githubusercontent.com/69414078/171715597-9e883d73-5cff-49bf-97b4-a48b2da559ed.png)

Picture 2:
![Modbus unit slave adr](https://user-images.githubusercontent.com/69414078/171719403-4126301c-5fe4-4925-880c-369ebbef7c8a.png)

Picture 3:
![Uploading Node red companion.png…]()

**Comment by mist0706** [comment]
So i had not installed the Node-Red companion. I did wonder how node-red communicated with HASS, it's actually not stated in the readme.

I'm not using a IAM module but instead a HF2211 modbusrtu to tcp wifi adapter. I've tested it with the homeassistant modbus module and it seems to work fine.

The sensors have now showed up (all 88) but it's still showing unknown on almost all of them. Not sure if i just have to wait a while or if there is something else that needs fixing.

My modbus RTU settings from the device itself is as following
Slavaddress: 1
Baudhastighet: 115200
Paritet: Ingen
Stoppbits: 1
Smartly-gateway: Ej ansluten
![Screenshot 2022-06-03 at 09 12 03](https://user-images.githubusercontent.com/10143260/171805901-1db5d426-efd4-4c66-b0e0-060cda4e94f8.png)

**Comment by GMTrevis** [comment]
> So i had not installed the Node-Red companion. I did wonder how node-red communicated with HASS, it's actually not stated in the readme.
> 
> I'm not using a IAM module but instead a HF2211 modbusrtu to tcp wifi adapter. I've tested it with the homeassistant modbus module and it seems to work fine.
> 
> The sensors have now showed up (all 88) but it's still showing unknown on almost all of them. Not sure if i just have to wait a while or if there is something else that needs fixing.
> 
> My modbus RTU settings from the device itself is as following Slavaddress: 1 Baudhastighet: 115200 Paritet: Ingen Stoppbits: 1 Smartly-gateway: Ej ansluten ![Screenshot 2022-06-03 at 09 12 03](https://user-images.githubusercontent.com/10143260/171805901-1db5d426-efd4-4c66-b0e0-060cda4e94f8.png)

You are right, i will add a description of the Node Red Companion in the Readme.

I assume you cannot read modbus data in node red either? (picture 1)
Have you configured the modbus client node in node red? (picture 2)

Im not sure but maybe the gateway your using cannot handle that much of modbus data as the IAM gateway. 
Could be tested by reading only a few modbus registers as a test, and not all.
I assume you was reading a lot less modbus registers in HA than you now are trying to read using Node red?
I guess you have done this but i'll mention it anyway. You cannot communate with the ventilation unit from both HA and Node Red at the same time, have you de-activated the modbus connection between HA and the ventilation unit?

Reducing the modbus baudrate could help, mine is at 9600 (i see your are given to 115200).

When i started the integration i was having trouble loosing connection to the ventilation unit when deploying changes from Node Red, and i had to restart the Node Red addon to re-establish connection. After changing deployment from "Full" to Modified Nodes", i haven't had any issues. (Picture 3).

I hope this helps out with your connection issues.

Picture 1:
![Node red](https://user-images.githubusercontent.com/69414078/171810972-e934f833-3f95-4e86-9584-0b69aed55357.png)

Picture 2:
![Modbus](https://user-images.githubusercontent.com/69414078/171811035-89d9e7f7-4bba-4be2-b53a-b6f2c969a6ac.png)

Picture 3:
![Deploy](https://user-images.githubusercontent.com/69414078/171813170-885589d3-1bbc-4b77-87ba-2aaecb8bd46f.png)

**Comment by mist0706** [comment]
So it seems i had missed to change the port from 502 to 8899, now most things seem to start working:
![Screenshot 2022-06-03 at 10 56 02](https://user-images.githubusercontent.com/10143260/171822833-4842bce6-7409-46c1-80d7-89d90b06e3e6.png)

The main reason for me to add this is to see the status of my freecooling. Or rather to understand the entire ecosystem in my house and make sure all functions are working as intended.

I was also wondering if the VSR500 with my 220mm tunnels are breaking the volume calculations?

Oh and thank you so much for the amazing help!

By the way i tried setting the temperature to 22 instead of 18 and it doesn't update on the SystemAir control panel or what it's called.

**Comment by GMTrevis** [comment]
> So it seems i had missed to change the port from 502 to 8899, now most things seem to start working: ![Screenshot 2022-06-03 at 10 56 02](https://user-images.githubusercontent.com/10143260/171822833-4842bce6-7409-46c1-80d7-89d90b06e3e6.png)
> 
> The main reason for me to add this is to see the status of my freecooling. Or rather to understand the entire ecosystem in my house and make sure all functions are working as intended.
> 
> I was also wondering if the VSR500 with my 220mm tunnels are breaking the volume calculations?
> 
> Oh and thank you so much for the amazing help!
> 
> By the way i tried setting the temperature to 22 instead of 18 and it doesn't update on the SystemAir control panel or what it's called.

Im happy to help, glad you figured it out :)
I would'nt exspect too much from the Freecooling feature, the airflow is too little to have any noticable effect. Maybe some effect but not much.

It's good to be able to monitor the functionality of such system and do proper adjustments.

Your 500 unit has more airflow than my VTR300, you have to change the "Calc. m3" function node in node red to corresond with your unit (it's just an estimate). Mine is max 300[m3/h], (Pådrag [%]*3) give 300m3/h when at 100% pådrag (Your factor should  probably be 5 if the unit give 500m3/h).

If you can read modbus data you should also be able to write modbus data, i assume the "Utekomp feature is off". If the utekomp feature is active, will that feature be in control of the setpoint and change the setpoint with the outdoor temperature (according to the setpoints in the pop-up).
 
![VTR300](https://user-images.githubusercontent.com/69414078/171849201-fb8e256b-7f36-4818-8db1-73f65fef71f9.png)
